CoreScriptsRoot/CoreScripts/ContextActionTouch.lua

```diff
@@ -171,6 +208,65 @@ function createNewButton(actionName, functionInfoTable)
 	return contextButton
 end
 
+function getButtonPosition(index, buttonSize)
+	local complete = false
+	local position, occupyingPosition
+
+	while not complete do
+		if buttonPositionsOccupied[index] then
+			index = index + 1
```

Comment:
It would be cleaner to move out the index search to a separate function named some like FindOpenPosition()